src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,47 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import core.mate.academy.service.impl.BulldozerProducerImpl;
+import core.mate.academy.service.impl.ExcavatorProducerImpl;
+import core.mate.academy.service.impl.TruckProducerImpl;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (type == Bulldozer.class) {
+            MachineProducer<Bulldozer> producer = new BulldozerProducerImpl();
+            return new ArrayList<>(producer.get());
+        } else if (type == Excavator.class) {
+            MachineProducer<Excavator> producer = new ExcavatorProducerImpl();
+            return new ArrayList<>(producer.get());
+        } else if (type == Truck.class) {
+            MachineProducer<Truck> producer = new TruckProducerImpl();
+            return new ArrayList<>(producer.get());
+        } else {
+            return Collections.emptyList();
+        }
```

Comment:
Consider creating a local variable for the `MachineProducer` to avoid repeated code when calling `producer.get()`. This will make the code cleaner and reduce duplication.